src/net_processing.cpp

```diff
@@ -2426,9 +2426,7 @@ void PeerLogicValidation::ProcessMessage(CNode& pfrom, const std::string& msg_ty
         UpdatePreferredDownload(pfrom, State(pfrom.GetId()));
         }
 
-        if (!pfrom.IsInboundConn() && pfrom.IsAddrRelayPeer())
-        {
-            // Advertise our address
+        if (!pfrom.IsInboundConn() && !pfrom.IsBlockOnlyConn()) {
```

Comment:
This is technically not a behavior change, but I don't understand the logic behind replacing `IsAddrRelayPeer` with `!IsBlockOnlyConn()`. Why not replacing it with `RelayAddrsWithConn`?

Reply:
I see it is justified (sort of?) [here](https://github.com/bitcoin/bitcoin/pull/19724#discussion_r474387266).

Reply:
The skipping of this logic for block-relay-only outbound peers was introduced in #15759 -- gating it on whether the address-relay data structures are instantiated is confusing because the very last thing we do is unrelated to whether those data structures exist.

Instead, what this block has in common is that we only want to skip it for outbound block-relay only peers, which is what this patch does.  So I think this is right, and while we could split this block into two parts, one gated on whether the peer has the addr-relay data structures, and another gated on whether the peers is an outbound block-relay only connection, that seems pretty unnecessary to me.  If someone adds another connection type that is outbound and non-block-relay only and only needs part of this, it can be refactored then.